### DISCUSSION TITLE
autorg errors and quality

### DISCUSSION THREAD

**Discussion by I-G-E** [biosaxs-com/atsas-community#1942] [Primary data processing] [answered]
Hello,
 
I am trying to use ATSAS 3.1.1 to analyse my SAXS data, starting with radius of gyration.
 
When I use the PRIMUS Guinier wizard with Autorg to analyse a dataset I get a value for Rg of 38.45 +/- 0.10 and a fidelity score of 0.84. Then when I use the terminal/command line version of autorg I get a value of Rg of 38.4 +/- 1.6 and a quality score of 91%. Another example with a different dataset: The PRIMUS Guinier Wizard with Autorg gives me a value for Rg of 37.42 +/- 0.12 and a fidelity score of 0.06. The command line version of autorg gives me a value of 37.4 +/- 0.2 and a quality score of 77%.
 
Both the PRIMUS GUI version of autorg and the command line version of autorg are using the same datapoint range to calculate Rg, so why am I getting different error values? Which is the recommended value to use?

I also can’t find any literature explaining how the fidelity score and quality score are calculated. I assume they both refer to the quality of the fit to the Guinier region but then why are there such discrepancies between the two scores? Can anyone clarify how are they calculated? And which ‘quality’ measure is the recommended value to report?

Thanks!

**Comment by franke-biosaxs** [comment, marked as answer]
If possible, please upgrade to [ATSAS-4.0](https://github.com/biosaxs-com/atsas-community/releases). That said, updating won't change what you observed.

> When I use the PRIMUS Guinier wizard with Autorg to analyse a dataset I get a value for Rg of 38.45 +/- 0.10 and a fidelity score of 0.84. Then when I use the terminal/command line version of autorg I get a value of Rg of 38.4 +/- 1.6 and a quality score of 91%. 

AUTORG includes extra information in the uncertainty term that is not present in the standard definition; the author described it thus:
> autorg provides "Rg StDev" which is "standard deviation of the experimental data from the Guinier fit in the selected interval plus standard deviation of Rg values from all possible intervals from the Rg value from the selected interval."

As AUTORG does not calculate the Rg/I0 values of any arbitrary range, primus recalculates the Rg and its uncertainty with DATRG for consistent display in the GUI. DATRG simply calculates the standard uncertainty of the regression slope.

> I also can’t find any literature explaining how the fidelity score and quality score are calculated. I assume they both refer to the quality of the fit to the Guinier region but then why are there such discrepancies between the two scores? Can anyone clarify how are they calculated? And which ‘quality’ measure is the recommended value to report?

DATRG reports the goodness-of-fit of the Guinier approximation based on a chi-square statistic. AUTORG has a more heuristic approach. You can find details on AUTORG in [section 3 of the 2007 ATSAS paper](https://doi.org/10.1107/S0021889807002853).

**Reply by I-G-E** [reply]
Thank you for the explanations. This has clarified most of my questions. However I am still a little confused at the discrepancies I have seen between fidelity and quality for certain datasets. For example, for one dataset I get a very low fidelity score calculated by primus/datrg (indicating a very poor Guinier fit), yet with the same dataset using the same range of datapoints for Guinier fit, I get a reasonably high quality score calculated by autorg command line (indicating a good quality of data and accurate Rg). Which score should I be reporting and should I be concerned by the low fidelity score if the quality score is high?

**Reply by franke-biosaxs** [reply]
> Which score should I be reporting and should I be concerned by the low fidelity score if the quality score is high?

As fidelity is a chi-square statistic, it not only depends on the residuals ($$I_{exp} - I_{fit}$$), but also the error estimates of the data. The quality of your error estimates will affect the fidelity. And the autorg score is, as said, a heuristic. 

For reporting, follow the [reporting guidelines](https://doi.org/10.1107/S2059798322012141).
